variables.tf

```diff
@@ -105,6 +105,12 @@ variable "bad_stream_name" {
   type        = string
 }
 
+variable "custom_paths" {
```

Comment:
Seems like a good addition but wondering if there should be some more input validation?  The default paths are fixed right (we have three options I believe) so should this instead be mapped as:

```
variable "custom_post_paths" {
  type = list(string)
  default = []
}
```

This would then map every entry to use the default-path of `/com.snowplowanalytics.snowplow/tp2`.  Worried that users could use an invalid default-path here without having this level of structure ... 

Would users ever want to define a custom default-path that is not one of the pre-existing ones we have mapped?

Reply:
It's not particularly clear to me how you would map the list to the specific path you wanted to override? The order of items in the list, the seems better for validation but not very description from a config point of view. I'm not wondering what the collector does it you use invalid paths, if it just ignores them then I don't see much risk in the lack of validation here... brb...

Reply:
A configuration like:

```
  paths {
    "/com.acme/track" = "/com.snowplow"
  }
```

doesn't have any negative impacts on the collector, it simply doesn't match anything so it ignore it. Same if the path on the left hand side is nonsense too, it simply gets ignored.

e.g.
```
  paths {
    "/com.acme/track" = "/com.snowplow"
    "fgxnbxfszgbfsxzg" = "/com.snowplowanalytics.iglu/v1"
  }
```

Given that this must be a string map, then I feel like its pretty safe, even if misconfigured with silly values.

Reply:
A little further digging into this... I think there are two approaches we could take:

We create three input options which are each a List, `custom_post_paths`, `custom_redirect_paths` and `custom_iglu_webhook_paths` and these map to the corresponding routes. This reduces the change of misconfiguring this option, and means users don't need to go digging to find the default paths.

Or we leave it as an object. This offers the most flexibility and allows people to configure the webhook paths and such too (`/com.mailchimp/v1` etc). However, these are _rarely_ configured as they are Server to Server requests typically and this feature mainly exists for web tracking.

We could also have both. Three lists for easy configuration, and then `additional_custom_paths` which we can merge into the object.

Reply:
Go for a map + enough docs to let people know when they might be doing something silly and that likely cuts the right mixture and keep this code simple.